src/WebJobs.Script.WebHost/Security/KeyManagement/DefaultSecretManagerProvider.cs

```diff
@@ -74,64 +74,56 @@ private void ResetSecretManager()
             Interlocked.Exchange(ref _secretManagerLazy, new Lazy<ISecretManager>(Create));
         }
 
-        private ISecretManager Create() => new SecretManager(CreateSecretsRepository(), _loggerFactory.CreateLogger<SecretManager>(), _metricsLogger, _hostNameProvider, _startupContextProvider);
+        private ISecretManager Create() => SecretsEnabled ? new SecretManager(CreateSecretsRepository(), _loggerFactory.CreateLogger<SecretManager>(), _metricsLogger, _hostNameProvider, _startupContextProvider) : null;
```

Comment:
Changes to this file shouldn't be needed

Reply:
Reverted.